core/src/main/java/org/apache/brooklyn/core/mgmt/BrooklynTags.java

```diff
@@ -59,6 +59,21 @@ public String getKind() {
         public String getContents() {
             return contents;
         }
+
+        @Override
+        public boolean equals(Object other) {
+            if (other == null || !(other instanceof NamedStringTag)) {
+                return false;
+            }
+
+            return kind != null && kind.equals(((NamedStringTag)other).kind)
```

Comment:
This will return not equals if both have `kind == null` and/or `contents == null`. Better to do:
```
if (!(other instanceof NamedStringTag)) {
    return false;
}
NamedStringTag o = (NamedStringTag) other;
return Objects.equal(kind, o.kind) && Objects.equal(contents, o.contents);
```